core/src/main/java/io/grpc/MethodDescriptor.java

```diff
@@ -262,4 +269,40 @@ public static String extractFullServiceName(String fullMethodName) {
     }
     return fullMethodName.substring(0, index);
   }
+
+  /**
+   * An internal only class for passing information directly into the transport.
+   */
+  @Internal
+  public static final class InternalAccessor {
+    private static final String PACKAGE = InternalAccessor.class.getPackage().getName();
+
+    private static final Set<String> PACKAGE_WHITELIST =
+        new HashSet<String>(Arrays.asList(PACKAGE + ".netty."));
+
+    /**
+     * Only packages on the whitelist may access this.
+     */
+    @Internal
+    public InternalAccessor() {
+      StackTraceElement[] st = new Throwable().getStackTrace();
```

Comment:
I'm sorta okay with letting this in as-is, but I really don't see why bothering in the first place. What is so much worse about this method than, say, `Metadata.serialize()`? I can understand the concern, but I don't see this as a _new_ concern. Thus caller-verification seems orthogonal to me, and I don't understand why I have to accept the caller enforcement to receive the performance optimization. I'm driven toward doing what we've done in the past: KISS and just have an `@Internal` getter.

I don't think we should be in the habit of adding things and removing them if they cause trouble. Instead we add things when they have been shown to be necessary. I see trying to merge this as-is for speed of review a false dichotomy.